client/engine/channellocker.go

```diff
@@ -0,0 +1,65 @@
+package engine
+
+import (
+	"bytes"
+	"sort"
+	"sync"
+
+	"github.com/statechannels/go-nitro/channel"
+	"github.com/statechannels/go-nitro/types"
+)
+
+// ChannelLocker is a utility class that allows for locking of channels to prevent concurrent updates to the same channels.
+type ChannelLocker struct {
+	channelLocks sync.Map
+}
+
+// NewChannelLocker returns a new ChannelLocker
+func NewChannelLocker() *ChannelLocker {
+	return &ChannelLocker{
+		channelLocks: sync.Map{},
+	}
+}
+
+// Lock acquires a locks on the given channels.
+func (l *ChannelLocker) Lock(channelIds []types.Destination) {
```

Comment:
I thought that `Lock` would _always_ be called with exactly two channels. Does it make sense to reflect that in the signature? This might also simplify the implementation.

Reply:
It depends on what role the participant is playing in a virtual channel. If the a participant is a hub they will have two ledger channels, otherwise they'll just have one ledger channel.

In our current implementation we also need to lock the virtual channel to prevent concurrently process messages for the same objective at the same time. This means we're locking either 2 or 3 channels when locking for a virtual fund objective.